core/client/src/main/java/org/phenotips/remote/client/internal/RemoteMatchFinder.java

```diff
@@ -78,6 +92,36 @@
     @Inject
     private ConsentManager consentManager;
 
+    @Inject
+    private Provider<XWikiContext> provider;
+
+    private XWikiDocument prefsDoc;
+
+    @Override
+    public void initialize() throws InitializationException
+    {
+        this.prefsDoc = this.getMatchingRunInfoDoc();
+    }
+
+    private XWikiDocument getMatchingRunInfoDoc()
+    {
+        try {
+            XWikiContext context = this.provider.get();
+            XWikiDocument doc = context.getWiki().getDocument(MATCHING_RUN_INFO_DOCUMENT, context);
+
+            if (doc != null && !doc.isNew()) {
+                return doc;
```

Comment:
what if `doc.isNew()`, whats wrong with that for this code?

Reply:
We use `doc == null || doc.isNew()` for `inaccessible or deleted document`